src/scss/_theme-helpers.scss

```diff
@@ -0,0 +1,96 @@
+// These mixins are build in a component agnostic way, this is why they set 
+// a component name. To break these mixins out into a reusable theming component 
+// rename to remove table references and introduce a `$component-name` argument.
+$_o-table-themes-avalible: () !default;
+
+
+// Set a new table theme.
+//
+// @include oTableThemeSetTheme('minimal', (
+// 	stripes: (
+// 		color: black,
+// 		alt-color: white,
+// 		background: white,
+// 		alt-background: black
+// 	)
+// ));
+//
+// @param {String} $theme - The name of the theme being configured, e.g. "masterbrand".
+// @param {Map} $theme-states - A map of theme states, a theme state is a key value map of configuration.
+@mixin oTableThemeSetTheme($theme, $theme-states) {
+	$component-name: "o-table";
+	// Get all themes for the provided component name.
+	$themes-for-component: map-get($_o-table-themes-avalible, $component-name);
+	// If no themes are yet set for this component create an empty map to contain them.
+	@if type-of($themes-for-component) != map {
+		$themes-for-component: ();
+	}
+	// Set the new states for this theme.
+	$themes-for-component: map-merge(
+		$themes-for-component,
+		($theme: (states: $theme-states))
+	);
+	// Merge the updated component themes.
+	$_o-table-themes-avalible: map-merge(
+		$_o-table-themes-avalible,
+		($component-name: $themes-for-component)
+	) !global;
+}
+
+// Get table theme configuration for a given state.
+//
+// @include oTableThemeGetFor('minimal', 'stripes');
```

Comment:
Not a huge issue but should the example for this (and `oTableThemeStateConfigured`) demonstrate that it's a function and not a mixin? E.g.

```scss
background-color: oTableThemeGetFor('minimal', 'stripes');
```

and


```scss
@if oTableThemeStateConfigured('minimal', 'stripes') {
    // do something
}
```